### PR TITLE
spdx headers

### PR DESCRIPTION
**Pull request by broskoTT**
Related to #47 
Adding pre-commit check for SPDX headers. 
Deprecates #185 

### API Changes
No API changes introduced

### CODE REVIEW

File: .check-copyright-config.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+DEFAULT:
```
Comment (broskoTT) [inline .check-copyright-config.yaml:1]: Will switch branch to point to main once the other PR is checked in

File: .check-copyright-config.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    # SPDX-License-Identifier: {license}
+
+  # comment lines matching:
+  # SPDX-FileCopyrightText: year[-year] Espressif Systems
+  # or
+  # SPDX-FileContributor: year[-year] Espressif Systems
+  # are replaced with this template prefixed with the correct comment notation (# or // or *) and SPDX- notation
+  espressif_copyright: '{years} Tenstorrent Inc.'
+
+# # You can create your own rules for files or group of files
```
Comment (pjanevskiTT) [inline .check-copyright-config.yaml:28]: nit: this is under comment twice

File: device/tt_io.hpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// SPDX-FileCopyrightText: (c) 2024 Tenstorrent Inc.
+// SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
```
Comment (joelsmithTT) [inline device/tt_io.hpp:1]: Why? This file didn't exist in 2023.

**Review by joelsmithTT — APPROVED** [review]

**Review by blozano-tt — APPROVED** [review]

File: device/tt_io.hpp (hunk 3 above)
Comment (broskoTT) [reply device/tt_io.hpp:1]: I don't know what is our policy on this. Just added 2023 everywhere.
 Should we update these each year? Or are they supposed to hold the year the file was first created?

Comment (joelsmithTT) [reply device/tt_io.hpp:1]: I think it's supposed to be the latter but I'm not sure.  Probably doesn't matter

**Pull request merged**